holoviews/plotting/bokeh/callbacks.py

```diff
@@ -1,446 +1,225 @@
+import json
 from collections import defaultdict
 
-import numpy as np
 import param
+import numpy as np
+from bokeh.models import CustomJS
 
-from ...core.data import ArrayColumns, PandasInterface
-from .util import compute_static_patch, models_to_json
-
-from bokeh.models import CustomJS, TapTool, ColumnDataSource
-from bokeh.core.json_encoder import serialize_json
-from bokeh.io import _CommsHandle
-from bokeh.util.notebook import get_comms
+from ...streams import (Stream, PositionXY, RangeXY, Selection1D, RangeX,
+                        RangeY, PositionX, PositionY, Bounds)
+from ..comms import JupyterCommJS
 
 
-class Callback(param.ParameterizedFunction):
+def attributes_js(attributes):
```

Comment:
In the end, I am not convinced that this class needs to be parameterized.

Lastly, it might be nice to define ``Callback`` to be backend agnostic. These methods look fairly independent from bokeh in their semantics to me:

```
def __init__(self, plot, streams, source, **params):
      
def initialize(self):

def on_msg(self, msg):

def _process_msg(self, msg):
```

No need to abstract it out in this PR though...